### DISCUSSION TITLE
How to setting ImGui in eclipse C/C++??

### DISCUSSION THREAD

**Discussion by Minster23** [eclipse-cdt/cdt#981] [Q&A] [answered]
**Hi**, i trying to use **ImGui** in eclipse, i followed the instructions from **ChatGPT**, but still **didnt** work

Can someone help?

**Comment by Minster23** [comment]
im using windows 11 and mingw

**Reply by jonahgraham** [reply]
Can you share which instructions you tried that didn't work please. Link to the steps you are following would be great.

**Reply by Minster23** [reply]
1. i set IMGUIand GLFW path -> Project -> Properties -> C/C++ General ->Path and symbol-> Includes =>(Path to imgui and imgui/backends
![image](https://github.com/user-attachments/assets/b4e54494-d86a-40ae-adc3-cb778880936e)

2. i add the path to linker too
![image](https://github.com/user-attachments/assets/a4473da8-cda4-430c-814c-64ae1521ea84)

3. **RESULT**
![image](https://github.com/user-attachments/assets/b271d965-efa4-468f-af79-6db36a3d7574)

**Reply by Minster23** [reply]
i import the imgui and is still error
![image](https://github.com/user-attachments/assets/58f306d0-08b6-49df-89eb-2057974a2d48)

**Reply by jonahgraham** [reply]
> i import the imgui and is still error
> ![image](https://github.com/user-attachments/assets/58f306d0-08b6-49df-89eb-2057974a2d48)
> 

This screenshot doesn't show any errors, a few warning and notes that should be taken into account possibly.

**Reply by Minster23** [reply]
it shows
g++: error: Gui\imgui: No such file or directory
g++: error: Gui\imgui\backends: No such file or directory

**Reply by Minster23** [reply]
i already add everything
but still error

**Reply by jonahgraham** [reply]
> i already add everything
> but still error
> 

Please post screenshot of the error you are seeing.

**Reply by jonahgraham** [reply]
My guess, but not sure, is that you need to quote the library search and include paths because your paths have spaces and Makefiles aren't very fond of spaces. I.e. Put the " in the text box around the include path, or move the Imgui to a path without spaces.

**Reply by Minster23** [reply]
it still error
![image](https://github.com/user-attachments/assets/437d2c9e-1a3e-4973-a51d-5cf6f67fd579)

**Reply by Minster23** [reply]
maybe i will change to Visual Studio

**Reply by jld01** [reply, marked as answer]
Hi @Minster23, the "undefined reference to ..." errors are from the GNU linker. You must add the _folders_ containing your ImGui and GLFW libraries to the _Library search path (-L)_ list and add the individual library _names_ to the _Libraries (-l)_ list. When adding a library name, you must omit the `lib` filename prefix and the `.a` filename suffix. For example, a library archive file named `libglfw3.a` should be listed as `glfw3` in the libraries list.

Does this explanation answer your question?
